gbn/timeout_manager.go

```diff
@@ -0,0 +1,277 @@
+package gbn
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+const (
+	defaultTMHandshakeTimeout = 100 * time.Millisecond
+	defaultTMResendTimeout    = 100 * time.Millisecond
+	minimumResendTimeout      = 100 * time.Millisecond
+	defaultTMFinSendTimeout   = 1000 * time.Millisecond
+	defaultResendMultiplier   = 5
+	DefaultTMSendTimeout      = math.MaxInt64
+	DefaultTMRecvTimeout      = math.MaxInt64
```

Comment:
perhaps just move the existing one's here? that way we dont need to rename them again later on.